sacrebleu.py

```diff
@@ -1296,9 +1296,9 @@ def compute_bleu(correct: List[int],
     smooth_mteval = 1.
     effective_order = NGRAM_ORDER
     for n in range(NGRAM_ORDER):
-        if smooth_method == 'add-k' and n > 1:
-            correct[n] += smooth_value
-            total[n] += smooth_value
```

Comment:
This edit is questionable because the comment [add-k: Method 2 (generalizing Lin and Och, 2004)](https://github.com/mjpost/sacreBLEU/blob/f9847a5/sacrebleu.py#L1281) suggests the intent was to generalize add-one smoothing to add-k.

However, I agree the current code is wrong because

* `smooth_value` defaults to `SMOOTH_VALUE_DEFAULT` which is 0.0, so by just using `--smooth=add-k` we don't get any smoothing, which is unexpected for the users.
* There is inconsistency whether `--smooth-value` applies to [`--smooth floor` only](https://github.com/mjpost/sacreBLEU/blob/f9847a5/sacrebleu.py#L73) or anytime [when relevant](https://github.com/mjpost/sacreBLEU/blob/f9847a5/sacrebleu.py#L1615).
* There is inconsistency between `add-k` and `add-n`. Currently, ['exp', 'floor', 'add-n', 'none'](https://github.com/mjpost/sacreBLEU/blob/f9847a5/sacrebleu.py#L1612) are the only allowed values of the `--smooth` parameter, but the rest of the code uses `add-k`, which I consider a better name (because n is used for the n-gram order). Fixing this bug is simple (renaming add-n to add-k), but we can see that no one actually uses this smoothing and it is not covered by the tests. So maybe we could rename it to `add-one` and forget about generalizing which no one uses.

Reply:
+1 to the issue with the --smooth parameter.

I would also adding a extra condition to 
https://github.com/mjpost/sacreBLEU/blob/f9847a52c6b447459a0475a5ae461963ff74ed82/sacrebleu.py#L1308-L1313
```
elif smooth_meth != "none": 
     raise ValueError(f"Invalid smoothing method {smooth_meth}") 
     #                   ^ remove f string python 2 compat
```
This would make sure typos in smooth_meth does not result in 'none' smoothing applied when expect 'add-n' or 'flopr' or whatever...

I also agree that having an explicit "add-one" is option is good option which less likely for the user to mess up by the default smooth-value. Either that or "--smooth-value" should be non-optional when using `--smooth add-k` .

Reply:
@martinpopel You are right. I renamed add-n to add-k and now add-k defaults to 1.
@DNGros I think the check is not necessary as argparse choices parameter restricts the allowed values and after fixing it by renaming add-n to add-k there is no room for typos anymore.

Reply:
I agree with the move from `add-n` to `add-k`. However, the intent of `--smooth-value` is to be a general value applicable to any smoothing method, and the current changes you proposed hard-code `add-k` to `1`. 

I suggest that we make `SMOOTH_VALUE_DEFAULT` a `dict` indexed by the smoothing methods. e.g.,

```python
SMOOTH_VALUE_DEFAULT = { 'floor': 0.0, 'add-k': 1 }
```
and so on.

Reply:
@sn1c that is a good point. However, it only applies if going through the CLI codepath. If you import sacrebleu and call compute_bleu from python code, I think there is a still a chance it might fail silently due to a typo. This is very minor, but in an ideal world it seems like that should be caught. 

(Don't let this stop this PR from going through though. This is a useful fix which hopefully can be merged soon so people's "add-n"/"add-k" aren't subtly wrong.)

Reply:
I think you've missed the point. If you do a diff from this branch against the current master, you'll see that the value of `smooth_value` is **entirely ignored** for `add-k`. In other words, `add-k` is hard-coded to `k=1`.